core/toolbox/separator.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A separator used for separating toolbox categories.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ * @author maribethb@google.com (Maribeth Bottorff)
+ */
+'use strict';
+
+goog.provide('Blockly.ToolboxSeparator');
+
+goog.require('Blockly.ToolboxItem');
+
+
+/**
+ * Class for a toolbox separator. This is the thin visual line that appears on
+ * the toolbox. This item is not interactable.
+ * @param {!Blockly.utils.toolbox.Separator} toolboxSeparatorDef The information
+ *     needed to create a separator.
+ * @param {!Blockly.IToolbox} toolbox The parent toolbox for the separator.
+ * @constructor
+ * @extends {Blockly.ToolboxItem}
+ */
+Blockly.ToolboxSeparator = function(toolboxSeparatorDef, toolbox) {
+
+  Blockly.ToolboxSeparator.superClass_.constructor.call(
+      this, toolboxSeparatorDef, toolbox);
+  /**
+   * The config for all the separator classes.
+   * @type {!Blockly.ToolboxSeparator.ClassConfig}
+   * @protected
+   */
+  this.classConfig_ = {
+    'container': 'blocklyTreeSeparator'
+  };
+
+  Blockly.utils.object.mixin(this.classConfig_,
+      toolboxSeparatorDef['classConfig']);
+};
+Blockly.utils.object.inherits(Blockly.ToolboxSeparator, Blockly.ToolboxItem);
+
+/**
+ * @typedef {{
+ *            container:?string,
```

Comment:
Is there a way that you can document what the properties of the typedef mean? This by itself doesn't explain what it is, it only provides typings.

Reply:
I don't know of a way to do this just yet. I have seen @property used in some places online, so I might be able to use that but will keep looking into it. If nothing else I'll add a line in the comment explaining what the container is.

Reply:
sgtm.